include/oneapi/mkl/dft/detail/dft_ct.hxx

```diff
@@ -0,0 +1,119 @@
+/*******************************************************************************
+* Copyright 2022 Codeplay Software
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+// Commit
+
+template <dft::detail::precision prec, dft::detail::domain dom>
+ONEMKL_EXPORT dft::detail::commit_impl *create_commit(dft::detail::descriptor<prec, dom> &desc);
+
+// BUFFER version
+
+//In-place transform
+template <typename descriptor_type, typename data_type>
+ONEMKL_EXPORT void compute_forward(descriptor_type &desc, sycl::buffer<data_type, 1> &inout);
+
+//In-place transform, using config_param::COMPLEX_STORAGE=config_value::REAL_REAL data format
+template <typename descriptor_type, typename data_type>
+ONEMKL_EXPORT void compute_forward(descriptor_type &desc, sycl::buffer<data_type, 1> &inout_re,
+                                   sycl::buffer<data_type, 1> &inout_im);
+
+//Out-of-place transform
+template <typename descriptor_type, typename input_type, typename output_type>
+ONEMKL_EXPORT void compute_forward(descriptor_type &desc, sycl::buffer<input_type, 1> &in,
```

Comment:
When using a reference to a descriptor (i.e. descriptor&), I'm hitting a situation where the only way to call the compute forward function is by doing something like this:

dft::compute_forward<std::remove_reference_t<decltype(descriptor)>, InputType, OutputType>(descriptor, in_dev, out_dev);

This is kinda ugly, and I'm hoping it could be possible to handle inside the template itself.

So I'm wondering if it would make sense to change the parameter to be a forwarding reference (i.e. descriptor_type&& desc). This, if I'm not mistaken, should allow it to bind to reference types as well. The reference would then be removed by std::forward.